app/controllers/units_controller.rb

```diff
@@ -0,0 +1,35 @@
+class UnitsController < ApplicationController
+  def create
+    @system = System.find(params[:unit][:system_id])
+    @unit = Unit.new(unit_params)
+    @unit.destination = @system
+
+    if @unit.save
+      flash[:notice] = "Unit Created"
+    else
+      flash[:notice] = "Failed to Create Unit"
+    end
+
+    respond_to do |format|
+      format.html { redirect_to system_path(@system) }
```

Comment:
It's supposed to run in AJAX, so the redirect should never really trigger in normal usage. Also, the only way for it to fail right now is for the form to be blank.